packages/design-system/src/Button/Button.stories.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+
+import { ButtonV2 as ButtonComponent } from "./Button";
+
+export default {
+  title: "Design System V2/Button",
+  component: ButtonComponent,
+  argTypes: {
+    as: {
+      control: {
+        type: "radio",
+        options: ["button", "a"],
+      },
+      description: "The HTML element to render the button as.",
+      table: {
+        type: { summary: "button | a" },
+        defaultValue: { summary: "button" },
+      },
+    },
+    className: {
+      control: {
+        type: "text",
+      },
+      description: "The class name to apply to the button component.",
+      table: {
+        type: { summary: "string" },
+      },
+    },
+    children: {
+      control: {
+        type: "object",
+      },
+      description: "The content to display within the button.",
+      table: {
+        type: { summary: "React.ReactNode | string" },
+      },
+    },
+    isLoading: {
+      control: {
+        type: "boolean",
+      },
+      description: "Whether the button should display a loading spinner.",
+      table: {
+        type: { summary: "boolean" },
+        defaultValue: { summary: "false" },
+      },
+    },
+    kind: {
+      control: {
+        type: "radio",
+        options: ["primary", "secondary", "tertiary", "error"],
+      },
+      description: "The visual style to apply to the button.",
+      table: {
+        type: { summary: "primary | secondary | tertiary | error" },
+        defaultValue: { summary: "primary" },
+      },
+    },
+    size: {
+      control: {
+        type: "radio",
+        options: ["sm", "md", "lg"],
```

Comment:
We should get the guidelines from Vasanth about the use cases for each type of button.